src/Commands/Utilities/REST/GraphHelper.cs

```diff
@@ -92,21 +92,17 @@ public static async Task<HttpResponseMessage> GetResponseAsync(PnPConnection con
         /// <param name="camlCasePolicy">Policy indicating the CamlCase that should be applied when mapping results to typed objects</param>
         /// <param name="propertyNameCaseInsensitive">Indicates if the response be mapped to the typed object ignoring different casing</param>
         /// <returns>List with objects of type T returned by the request</returns>
-        public static async Task<IEnumerable<T>> GetResultCollectionAsync<T>(PnPConnection connection, string url, string accessToken, bool camlCasePolicy = true, bool propertyNameCaseInsensitive = false)
+        public static async Task<IEnumerable<T>> GetResultCollectionAsync<T>(PnPConnection connection, string url, string accessToken, bool camlCasePolicy = true, bool propertyNameCaseInsensitive = false, IDictionary<string, string> additionalHeaders = null)
         {
             var results = new List<T>();
-            var request = await GetAsync<RestResultCollection<T>>(connection, url, accessToken, camlCasePolicy, propertyNameCaseInsensitive);
+            var request = await GetAsync<RestResultCollection<T>>(connection, url, accessToken, camlCasePolicy, propertyNameCaseInsensitive, additionalHeaders);
 
             if (request.Items.Any())
             {
                 results.AddRange(request.Items);
                 while (!string.IsNullOrEmpty(request.NextLink))
                 {
-                    request = await GetAsync<RestResultCollection<T>>(connection, request.NextLink, accessToken, camlCasePolicy, propertyNameCaseInsensitive);
-                    if (request.Items.Any())
-                    {
-                        results.AddRange(request.Items);
-                    }
+                    request = await GetAsync<RestResultCollection<T>>(connection, request.NextLink, accessToken, camlCasePolicy, propertyNameCaseInsensitive, additionalHeaders);
```

Comment:
Any reason you removed this portion ?

```
if (request.Items.Any())
                    {
                        results.AddRange(request.Items);
                    }
```

Reply:
Unintentional change left over from debugging - thanks for catching it :D I'll restore it.

Reply:
Oversight on my part - returning it. 😅